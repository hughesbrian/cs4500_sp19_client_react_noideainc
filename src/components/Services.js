import React from 'react'

const Services = ({services, editingService, setEditingService, deleteService,
                   createService, updateService, saveService}) =>
    <div>
        <h3>Services</h3>

        <table className="table">
            <tbody>
            <tr>
                <td><input onChange={(e) =>
                    setEditingService({
                        editingService: {
                            id: editingService.id,
                            title: e.target.value,
                            description: editingService.description
                        }
                    })} className="form-control title-edit-field"
                           value={editingService.title}/></td>
                <td><input onChange={(e) =>
                    setEditingService({
                        editingService: {
                            id: editingService.id,
                            title: editingService.title,
                            description: e.target.value
                        }
                    })} className="form-control"
                           value={editingService.description}/></td>

                <td>
                    <button className="btn btn-success save-service-btn" onClick={() => {
                        saveService(editingService, createService, updateService)
                    }}>Save
                    </button>
                    <button className="btn btn-warning update-service-button" onClick={() => {
                        setEditingService({
                            editingService: {
                                title: '',
                                description: ''
                            }
                        })
                    }}>Clear
                    </button>
                </td>
            </tr>
            {
                services.map(service =>
                    <tr key={service.id}>
                        <td>{service.title}</td>
                        <td>{service.description}</td>
                        <td>
                            <button className="btn btn-primary edit-service-btn" onClick={() => {
                                editingService = service
                            }}
                            >Edit
                            </button>
                            <button className="btn btn-danger delete-service-btn" onClick={
                                () => deleteService(service)
                            }>Delete
                            </button>
                        </td>
                    </tr>
                )
            }
            </tbody>
        </table>
    </div>;

export default Services
