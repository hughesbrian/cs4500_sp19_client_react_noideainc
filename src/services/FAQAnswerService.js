const baseURL = process.env.NODE_ENV !== 'development' ? 'http://localhost:8080' : 'https://cs4500-sp19-noideainc.herokuapp.com'

export default class FAQAnswerService {
    static instance = null;
    static getInstance() {
        if(FAQAnswerService.instance === null) {
            FAQAnswerService.instance = new FAQAnswerService()
        }
        return this.instance
    }

    findFAQAnswerById = id =>
        fetch(baseURL + `/api/faq-answers/${id}`)
            .then(response => response.json())

    findAllFAQAnswers = () =>
        fetch(baseURL + "/api/faq-answers")
            .then(response => response.json())

    createFAQAnswer = newFAQAnswer =>
        fetch(baseURL + "/api/faq-answers", 
        {
            method: 'POST',
            body: JSON.stringify(newFAQAnswer),
            headers: {'content-type': 'application/json'}
        }).then(response => response.json())

    deleteFAQAnswer = id =>
        fetch(baseURL + `/api/faq-answers/${id}`, 
        {
            method: 'DELETE'
        })

    editFAQAnswer = answer =>
        fetch(baseURL + `/api/faq-answers/${answer.id}`,
        {
            method: 'PUT',
            body: JSON.stringify({
                answer: answer.answer
            }),
            headers: {'content-type': 'application/json'}
        }).then(response => response.json())

    findFiltered = (answer) =>
        fetch(baseURL + "/api/faq-answers/filtered?answer=" + answer)
            .then(response => response.json())
}