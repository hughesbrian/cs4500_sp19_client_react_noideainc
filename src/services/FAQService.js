
const baseURL = process.env.NODE_ENV !== 'development' ? 'http://localhost:8080' : 'https://cs4500-sp19-noideainc.herokuapp.com'

export default class FAQService {
    static instance = null;
    static getInstance() {
        if(FAQService.instance === null) {
            FAQService.instance = new FAQService()
        }
        return this.instance
    }

    findFAQById = id =>
        fetch(baseURL + `/api/faqs/${id}`)
            .then(response => response.json())

    findAllFAQs = () =>
        fetch(baseURL + "/api/faqs")
            .then(response => response.json())

    createFAQ = newFAQ =>
        fetch(baseURL + "/api/faqs", 
        {
            method: 'POST',
            body: JSON.stringify(newFAQ),
            headers: {'content-type': 'application/json'}
        }).then(response => response.json())

    deleteFAQ = id =>
        fetch(baseURL + `/api/faqs/${id}`, 
        {
            method: 'DELETE'
        })

    editFAQ = question =>
        fetch(baseURL + `/api/faqs/${question.id}`,
        {
            method: 'PUT',
            body: JSON.stringify({
                question: question.question,
                title: question.title
            }),
            headers: {'content-type': 'application/json'}
        }).then(response => response.json())

    findFiltered = (title, question) =>
        fetch(baseURL + "/api/faqs/filtered?title=" + title + "&question=" + question)
            .then(response => response.json())

    findPagedFAQs = (page, count) =>
        fetch(baseURL + "/api/faqs/paged?page=" + page + "&count=" + count)
            .then(response => response.json())
}