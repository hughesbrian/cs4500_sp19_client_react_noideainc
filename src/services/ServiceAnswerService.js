export default class ServiceAnswerService {
    static instance = null;
    static getInstance() {
        if(ServiceAnswerService.instance === null) {
            ServiceAnswerService.instance = new ServiceAnswerService()
        }
        return this.instance
    }
    findServiceAnswerById = id =>
        fetch(`https://cs4500-sp19-noideainc.herokuapp.com/api/service-answers/${id}`)
            .then(response => response.json())
    findAllServiceAnswers = () =>
        fetch("https://cs4500-sp19-noideainc.herokuapp.com/api/service-answers/")
            .then(response => response.json())

    createServiceAnswer = answer => {
        delete answer.id;
        return fetch(`https://cs4500-sp19-noideainc.herokuapp.com/api/service-answers/`, {
            method: 'post',
            body: JSON.stringify(answer),
            headers: {
                'content-type': 'application/json'
            }
        })
            .then(response => response.json())
    }

    updateServiceAnswer= answer =>
        fetch(`https://cs4500-sp19-noideainc.herokuapp.com/api/service-answers/${answer.id}`, {
            method: 'put',
            body: JSON.stringify(answer),
            headers: {
                'content-type': 'application/json'
            }
        }).then(response => response.json())

    deleteServiceAnswer = id =>
        fetch(`https://cs4500-sp19-noideainc.herokuapp.com/api/service-answers/${id}`, {
            method: 'delete'
        })
}