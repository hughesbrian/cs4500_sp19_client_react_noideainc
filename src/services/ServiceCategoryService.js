const baseURL = "https://cs4500-sp19-noideainc.herokuapp.com";
//const baseURL = "http://localhost:8080";

export default class ServiceCategoryService {
    static instance = null;
    static getInstance() {
        if (ServiceCategoryService.instance === null) {
            ServiceCategoryService.instance = new ServiceCategoryService()
        }
        return this.instance
    }

    findServiceCategoryById = categoryId =>
        fetch(baseURL + `/api/categories/${categoryId}`)
            .then(response => response.json())

    findAllServiceCategories = () =>
        fetch(baseURL +  `/api/categories`)
            .then(response => response.json())

    pagedServiceCategories = (page, count) =>
        fetch(baseURL + `/api/categories/paged?page=${page}&count=${count}`)
            .then(response => response.json())

    createServiceCategory = sc => {
        delete sc.id;
        return fetch(baseURL + `/api/categories`, {
            method: 'post',
            body: JSON.stringify(sc),
            headers: {
                'content-type': 'application/json'
            }
        })
        .then(response => response.json())
    }

    updateServiceCategory = sc =>
        fetch(baseURL + `/api/categories/${sc.id}`, {
            method: 'put',
            body: JSON.stringify(sc),
            headers: {
                'content-type': 'application/json'
            }
        }).then(response => response.json())
    
    updateServiceCategoryScore = sc =>
        fetch(baseURL + `/api/categories/score/${sc.id}`, {
            method: 'put',
            body: JSON.stringify(sc),
            headers: {
                'content-type': 'application/json'
            }
        }).then(response => response.json())

    deleteServiceCategory = id =>
        fetch(baseURL + `/api/categories/${id}`, {
            method: 'delete'
        })

    findAllServicesByCategoryName = catName =>
        fetch(baseURL +  `/api/${catName}`)
            .then(response => response.json())
}
