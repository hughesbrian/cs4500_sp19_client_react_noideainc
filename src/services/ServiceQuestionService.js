export default class ServiceQuestionService {
    static instance = null;

    backend_url = 'http://localhost:8080';
    //backend_url = 'https://cs4500-sp19-noideainc.herokuapp.com';

    static getInstance() {
        if(ServiceQuestionService.instance === null) {
            ServiceQuestionService.instance = new ServiceQuestionService()
        }
        return this.instance
    }

    findServiceQuestionById = (id) =>
        fetch(`${this.backend_url}/api/service-questions/${id}`)
            .then(response => response.json())
    findAllServiceQuestions = () =>
        fetch(`${this.backend_url}/api/service-questions`)
            .then(response => response.json())

    findAllServiceQuestionsByServiceId = (serviceId) =>
        fetch(`${this.backend_url}/api/service-questions/service/${serviceId}`)
            .then(response => response.json())

    createQuestion = (question) =>
        fetch(`${this.backend_url}/api/service-questions`,
        {
            method: 'POST',
            body: JSON.stringify(question),
            headers: {'content-type': 'application/json'}
        })

    updateQuestion = (id, question) =>
        fetch(`${this.backend_url}/api/service-questions/${id}`,
        {
            method: 'PUT',
            body: JSON.stringify(question),
            headers: {'content-type': 'application/json'}
        })

    deleteQuestion = (id) =>
        fetch(`${this.backend_url}/api/service-questions/${id}`,
        {
            method: 'DELETE',
        })
}