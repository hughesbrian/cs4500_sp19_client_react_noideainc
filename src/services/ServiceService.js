export default class ServiceService {
    static instance = null;
    static getInstance() {
        if(ServiceService.instance === null) {
            ServiceService.instance = new ServiceService()
        }
        return this.instance
    }
    findServiceById = serviceId =>
        fetch("https://cs4500-sp19-noideainc.herokuapp.com/api/services/${serviceId}")
            .then(response => response.json())
    findAllServices = () =>
        fetch("https://cs4500-sp19-noideainc.herokuapp.com/api/services")
            .then(response => response.json())
    createService = service =>
        fetch(`https://cs4500-sp19-noideainc.herokuapp.com/api/services`,
            {
                method: 'POST',
                body: JSON.stringify(service),
                headers: {'content-type': 'application/json'}
            })
            .then(response => response.json())
    updateService = service =>
        fetch(`https://cs4500-sp19-noideainc.herokuapp.com/api/services/${service.id}`,
            {
                method: 'PUT',
                body: JSON.stringify(service),
                headers: {'content-type': 'application/json'}
            })
            .then(response => response.json())
    deleteService = service =>
        fetch(`https://cs4500-sp19-noideainc.herokuapp.com/api/services/${service.id}`,
            {
                method: 'DELETE',
            })

    updateServiceScore = service =>
        // change to http://localhost:8080 to test locally
        fetch(`hhttps://cs4500-sp19-noideainc.herokuapp.com/api/services/score/${service.id}`, {
            method: 'put',
            body: JSON.stringify(service),
            headers: {
                'content-type': 'application/json'
            }
        }).then(response => response.json())
}
