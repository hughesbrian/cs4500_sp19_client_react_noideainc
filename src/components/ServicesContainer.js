import React from 'react';
import Services from './Services';
import ServiceService from "../services/ServiceService";

class ServicesContainer extends React.Component {
    constructor(props) {
        super(props);
        this.serviceService = ServiceService.getInstance();
        this.state = {
            editingService: {
                title: '',
                description: ''
            },
            services: []
        }
    }

    componentDidMount() {
        this.serviceService
            .findAllServices()
            .then(services =>
                this.setState({
                    services: services
                })
            )
    }

    createService = (service) =>
        this.serviceService
            .createService(service)
            .then(this.props.history.push("/admin/services/"));

    saveService = function (service, updateService, createService) {
        if (service.id) {
            updateService(service);
        } else {
            createService(service);
        }
    };

    setEditingService = (new_attrs) =>
        this.setState(new_attrs);

    updateService = (service) =>
        this.userService
            .updateService(service)
            .then(this.props.history.push("/admin/services/"));

    deleteService = (service) =>
        this.userService
            .deleteService(service)
            .then(this.props.history.push("/admin/services/"));

    render = () => <Services services={this.state.services}
                          editingService={this.state.editingService}
                          setEditingService={this.setEditingService}
                          createService={this.createService}
                          updateService={this.updateService}
                          deleteService={this.deleteService}
                          saveService={this.saveService}/>;
}

export default ServicesContainer