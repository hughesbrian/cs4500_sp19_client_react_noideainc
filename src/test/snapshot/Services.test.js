import React from 'react'
import Services from '../../components/Services'
import TestRenderer from 'react-test-renderer'
import services from '../mockdata/Services.mock'
import ServiceService from '../../services/ServiceService'
import '../mockservice/ServiceService.mock'
import ServicesContainer from "../../components/ServicesContainer";

const serviceService = ServiceService.getInstance();

test('Create and Delete Services', () => {
    let clean3 = {id: 779, title: "cleaning3", description: "no3cleaningService"};

    const saveService = () => {
        services.push(clean3);

        let tree = testRenderer.toJSON();
        expect(tree).toMatchSnapshot();

    };

    const deleteService = serviceId => {
        services.splice(0, 1)

        let tree = testRenderer.toJSON();
        expect(tree).toMatchSnapshot();

    };

    const testRenderer = TestRenderer.create(
        <Services services={services}
                  editingService={clean3}
                  deleteService={deleteService}
                  saveService={saveService}/>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    const testInstance = testRenderer.root;

    const createServiceBtn = testInstance.findByProps({className: 'btn btn-success save-service-btn'});
    createServiceBtn.props.onClick();

    const deleteServiceBtns = testInstance.findAllByProps({className: 'btn btn-danger delete-service-btn'});
    deleteServiceBtns[1].props.onClick();
});

test('List of Services Renders Directly', () => {
    const testRenderer = TestRenderer.create(
        <Services services={services} editingService={services[0]}/>
    );

    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    const testInstance = testRenderer.root;

    const servicenameField = testInstance.findByProps({className: 'form-control title-edit-field'});
    const numServices = testInstance.findAllByProps({className: 'btn btn-danger delete-service-btn'});

    expect(servicenameField.props.value).toBe('cleaning2');
    expect(numServices.length).toBe(services.length);
});

test('List of Services Renders from Service', () => {
    serviceService.findAllServices().then(services => {
        const testRenderer = TestRenderer.create(
            <Services services={services} editingService={services[0]}/>
        );

        let tree = testRenderer.toJSON();
        expect(tree).toMatchSnapshot();

        const testInstance = testRenderer.root;

        const servicenameField = testInstance.findByProps({className: 'form-control title-edit-field'});
        const numServices = testInstance.findAllByProps({className: 'btn btn-danger delete-service-btn'});

        expect(servicenameField.props.value).toBe('cleaning1');
        expect(numServices.length).toBe(services.length);
    });
});
