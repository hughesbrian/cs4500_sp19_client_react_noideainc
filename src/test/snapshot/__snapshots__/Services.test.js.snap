// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create and Delete Services 1`] = `
<div>
  <h3>
    Services
  </h3>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <input
            className="form-control title-edit-field"
            onChange={[Function]}
            value="cleaning3"
          />
        </td>
        <td>
          <input
            className="form-control"
            onChange={[Function]}
            value="no3cleaningService"
          />
        </td>
        <td>
          <button
            className="btn btn-success save-service-btn"
            onClick={[Function]}
          >
            Save
          </button>
          <button
            className="btn btn-warning update-service-button"
            onClick={[Function]}
          >
            Clear
          </button>
        </td>
      </tr>
      <tr>
        <td>
          cleaning1
        </td>
        <td>
          p4ssw0rd
        </td>
        <td>
          <button
            className="btn btn-primary edit-service-btn"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="btn btn-danger delete-service-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          cleaning2
        </td>
        <td>
          pa55word
        </td>
        <td>
          <button
            className="btn btn-primary edit-service-btn"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="btn btn-danger delete-service-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Create and Delete Services 2`] = `
<div>
  <h3>
    Services
  </h3>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <input
            className="form-control title-edit-field"
            onChange={[Function]}
            value="cleaning3"
          />
        </td>
        <td>
          <input
            className="form-control"
            onChange={[Function]}
            value="no3cleaningService"
          />
        </td>
        <td>
          <button
            className="btn btn-success save-service-btn"
            onClick={[Function]}
          >
            Save
          </button>
          <button
            className="btn btn-warning update-service-button"
            onClick={[Function]}
          >
            Clear
          </button>
        </td>
      </tr>
      <tr>
        <td>
          cleaning1
        </td>
        <td>
          p4ssw0rd
        </td>
        <td>
          <button
            className="btn btn-primary edit-service-btn"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="btn btn-danger delete-service-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          cleaning2
        </td>
        <td>
          pa55word
        </td>
        <td>
          <button
            className="btn btn-primary edit-service-btn"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="btn btn-danger delete-service-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Create and Delete Services 3`] = `
<div>
  <h3>
    Services
  </h3>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <input
            className="form-control title-edit-field"
            onChange={[Function]}
            value="cleaning3"
          />
        </td>
        <td>
          <input
            className="form-control"
            onChange={[Function]}
            value="no3cleaningService"
          />
        </td>
        <td>
          <button
            className="btn btn-success save-service-btn"
            onClick={[Function]}
          >
            Save
          </button>
          <button
            className="btn btn-warning update-service-button"
            onClick={[Function]}
          >
            Clear
          </button>
        </td>
      </tr>
      <tr>
        <td>
          cleaning1
        </td>
        <td>
          p4ssw0rd
        </td>
        <td>
          <button
            className="btn btn-primary edit-service-btn"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="btn btn-danger delete-service-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          cleaning2
        </td>
        <td>
          pa55word
        </td>
        <td>
          <button
            className="btn btn-primary edit-service-btn"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="btn btn-danger delete-service-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`List of Services Renders Directly 1`] = `
<div>
  <h3>
    Services
  </h3>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <input
            className="form-control title-edit-field"
            onChange={[Function]}
            value="cleaning2"
          />
        </td>
        <td>
          <input
            className="form-control"
            onChange={[Function]}
            value="pa55word"
          />
        </td>
        <td>
          <button
            className="btn btn-success save-service-btn"
            onClick={[Function]}
          >
            Save
          </button>
          <button
            className="btn btn-warning update-service-button"
            onClick={[Function]}
          >
            Clear
          </button>
        </td>
      </tr>
      <tr>
        <td>
          cleaning2
        </td>
        <td>
          pa55word
        </td>
        <td>
          <button
            className="btn btn-primary edit-service-btn"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="btn btn-danger delete-service-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          cleaning3
        </td>
        <td>
          no3cleaningService
        </td>
        <td>
          <button
            className="btn btn-primary edit-service-btn"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="btn btn-danger delete-service-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`List of Services Renders from Service 1`] = `
<div>
  <h3>
    Services
  </h3>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <input
            className="form-control title-edit-field"
            onChange={[Function]}
            value="cleaning2"
          />
        </td>
        <td>
          <input
            className="form-control"
            onChange={[Function]}
            value="pa55word"
          />
        </td>
        <td>
          <button
            className="btn btn-success save-service-btn"
            onClick={[Function]}
          >
            Save
          </button>
          <button
            className="btn btn-warning update-service-button"
            onClick={[Function]}
          >
            Clear
          </button>
        </td>
      </tr>
      <tr>
        <td>
          cleaning2
        </td>
        <td>
          pa55word
        </td>
        <td>
          <button
            className="btn btn-primary edit-service-btn"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="btn btn-danger delete-service-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          cleaning3
        </td>
        <td>
          no3cleaningService
        </td>
        <td>
          <button
            className="btn btn-primary edit-service-btn"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="btn btn-danger delete-service-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
